### ISSUE TITLE
support get_json_array

### ISSUE THREAD

**Issue by INNOCENT-BOY** [StarRocks/starrocks#52616]
{
    "people": [{
        "name": "Daniel",
        "surname": "Smith"
    }, {
        "name": "Lily",
        "surname": "Smith",
        "active": true
    }]
}
For example, I have a JSON field in this format, and I want to query for rows where the user's name is "jack." However, there is no function like get_json_array available. Although I can extract the result using $[*].name, there's currently no way to convert this result into an array.
If the community finds this reasonable, I might try implementing this method, get_json_array. @wangsimo0

**Comment by INNOCENT-BOY** [comment]
Note: This issue is related to https://github.com/StarRocks/starrocks/issues/52585.

**Comment by wangsimo0** [comment]
Thanks! could you pls give a demo and description about how you want to implement this? like the function's name, input, output and examples. just want to make sure we are on the same page. also some reference will be even better.
here is an example  https://github.com/StarRocks/starrocks/issues/36245

**Comment by jaogoy** [comment]
Use `$.people[*].name` not `$[*].name`.

```SQL
select js->'$.people[*].name'
from (
    select parse_json('
    {
        "people": [{
            "name": "Daniel",
            "surname": "Smith"
        }, {
            "name": "Lily",
            "surname": "Smith",
            "active": true
        }]
    }') as js) t
```

**Comment by INNOCENT-BOY** [comment]
Thanks @jaogoy . Your guide have solved my problems. And one more question, don't you know some function to solve below issue?
I have two array: array1: [1, 2, 3], array2: [2, 3]
Check if array1 contains any element from array2.
We all know array_contains(param1, param2), but param2 could't be array.

**Comment by INNOCENT-BOY** [comment]
array_length(array_intersect) could solve my problem.
Do you have more smarter method? @jaogoy

**Comment by jaogoy** [comment]
> array_length(array_intersect) could solve my problem. Do you have more smarter method? [@jaogoy](https://github.com/jaogoy)

No, it needs such a function: https://github.com/StarRocks/starrocks/issues/58936
